compiler/rustc_mir_transform/src/simplify.rs

```diff
@@ -632,6 +650,16 @@ impl<'tcx> Visitor<'tcx> for UsedLocals {
             self.use_count[local] -= 1;
         }
     }
+
+    fn visit_var_debug_info(&mut self, var_debug_info: &VarDebugInfo<'tcx>) {
```

Comment:
Another idea here: I think we have a spare byte in `VarDebugInfo`, so we could put the [`DebugInfo`](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_session/config/enum.DebugInfo.html) level on each of these.

That would let us be aware of the original debuginfo level from the original code even when it's inlined, rather than depending on the level from the current crate.

Reply:
Now that we have SROA in MIR, we may need to check `VarDebugInfoContents::Composite` too.